### DISCUSSION TITLE
Sql generated with additional Chinese

### DISCUSSION THREAD

**Discussion by njalan** [vanna-ai/vanna#369] [Q&A]
Sometime there are Chinese at the end of Sql :
![image](https://github.com/vanna-ai/vanna/assets/8622340/f880c571-b80e-4c2d-9640-e118260d85fb)

**Comment by zainhoda** [comment]
@njalan could you provide some additional context here on the setup (LLM, training data, etc)

**Comment by njalan** [comment]
@zainhoda I am using Baichuan-7B as LLM and training consists of querion-sql pairs and documentation for business knowledge(it is mixed with Chinese and englist.) All the question are asked by Chinese

**Comment by njalan** [comment]
Sometimes it provided me the same two query splitted by -- 或者(it means OR in Englist) So is any any prompt to avoid it?

**Comment by zainhoda** [comment]
I think the easiest path might be to override the `extract_sql` method:

https://github.com/vanna-ai/vanna/blob/main/src/vanna/base/base.py#L126-L150

You can provide your own `extract_sql` method that will remove the unnecessary characters

**Comment by njalan** [comment]
@zainhoda Can you please help to fix it? I am not familiar with Python/

**Comment by zyclove** [comment]
@zainhoda  Can you support this feature? Looking forward to releasing it

**Reply by zainhoda** [reply]
What issue are you experiencing? Are you using the same LLM as the other user?

**Reply by zyclove** [reply]
@zainhoda We are developers in China. We hope that the input description is in Chinese and the think prompt output is also in Chinese. It would be better to support multiple languages.

**Comment by Mioooooo** [comment]
@zyclove Replace LLM and use qwen:32b or llama3:70b

**Comment by njalan** [comment]
When I use codeQwen1.5 and then didn't face this issue any more by now
